src/service/blocklist.py

```diff
@@ -24,3 +33,69 @@ def list_blocklist(self, user_id: str, list_type: BlockListType) -> ListBlockLis
         ]
 
         return ListBlockListResponse(blocklist=blocklist, status=ResponseStatus.SUCCESS)
+
+    def add_blocklist(self, user_id: str, domain: str, list_type: BlockListType) -> EditBlockListResponse:
+        """Add an url to blocklist."""
+        collection = self.db.get_collection("blocklist")
+
+        # Validate the website URL
+        if not URL_REGEX.match(domain):
+            return EditBlockListResponse(
+                status=ResponseStatus.FAILED,
+                user_id=user_id,
+                domain=domain,
+                list_type=list_type
+            )
```

Comment:
```suggestion
        # Validate the website URL
        if not URL_REGEX.match(domain):
            return EditBlockListResponse(
                status=ResponseStatus.FAILED,
                user_id=user_id,
                domain=domain,
                list_type=list_type
            )
            
        collection = self.db.get_collection("blocklist")
```

validating always be the first